### DISCUSSION TITLE
For a cleaner interface...

### DISCUSSION THREAD

**Discussion by TransLucida** [Hofknecht/SystemTrayMenu#380] [Q&A] [answered]
Hi, I design user interfaces for a living and I've been finding this tool super useful, but I have a few suggestions:

- Allow user to hide the search bar
- Allow user to hide the bottom icons
- Make bottom icons much smaller in comparison to the other icons
- Remove or allow user to hide the item count at the bottom
- Increase the margins around the menu so the icons don't stay too close to the edges
- Hide the curved arrows Windows adds to shortcuts
- Allow user to activate the menu with a right-click
- Allow user to let the program check for updates
- On launch make it look for the settings folder within its install folder first (so settings can survive Windows reinstalls)

Thanks

**Comment by Hofknecht** [comment, marked as answer]
@TransLucida 
Thank you for all that great ideas!

here are our thoughts on the topics, we will think about it again in the future and create issues

- [x] Allow user to hide the search bar 
=> sounds good, we will probably create an issue 
#386
- [x] Allow user to hide the bottom icons
=> sounds good, we need to think more about it
pin menu and open folder are not there in other ways
in the past we had open menu when double click title, maybe we could revive the idea, not sure right now 
#386
- [x] Make bottom icons much smaller in comparison to the other icons
=> sounds good, we will probably create a issue
the problem was that it should also be able to be pressed with touch, but we could distinguish between touch and non-touch here, or make it adjustable in some way
#386
- [x]  Remove or allow user to hide the item count at the bottom
=> sounds good, we will probably create a issue
#386
- [x] Increase the margins around the menu so the icons don't stay too close to the edges
=> sounds good, we will probably create a issue
#386
we wanted it to be as brief as possible or like the windows toolbar, but you're probably right here or at least have it adjustable
- [x] Hide the curved arrows Windows adds to shortcuts
=> That's a bigger topic, we had in past without these overlay arrows, so we have to think again or I think that the application folder doesn't have these arrows natively, we'll think again. the problem would simply be that you can't distinguish a shortcut from a file. possibly an option, but the implementation had certain challenges at the point
#394
- [x] Allow user to activate the menu with a right-click
Do you mean the taskbar item (not sure if this is possible because is already occupied with windows function ) or the system tray menu icon (there we show already the context menu) ?
=> rejected for now. Right click already occupied. function keys with a right click could be discussed again separately, we are currently not sure how that should look exactly.
- [x] Allow user to let the program check for updates
=> this is probably: 
#185 and/or 
#331
- [x] On launch make it look for the settings folder within its install folder first (so settings can survive Windows reinstalls)
=> we need to analyze this. Do you mean reinstall the app via Microsoft Store? 
 it should be like this, e.g. in portable version maybe works. if this is not the case it is a bug. 
=> we are currently unable to reproduce an error. The config is transferred to the next version. But it is not taken over if you have the option 'Save configuration in application directory' on. Also if we rename optionflags it is not transferred to the next version. You can create an issue here if this happens again. There it is best to specify from which to which version, then we can analyze and explain in detail.

**Comment by chip33** [comment]
Having the option to hide the title bar would be nice too and give a little bit more height for the folder items.

**Comment by Hofknecht** [comment]
related to
"Allow user to hide the search bar"
"Allow user to hide the bottom icons"
"Having the option to hide the title bar would be nice too"
find issue here: https://github.com/Hofknecht/SystemTrayMenu/issues/386

**Comment by Hofknecht** [comment]
@chip33 
related to "give a little bit more height for the folder items"
to adjust height of items there is a option here
![2022-05-27 22_33_31-Settings](https://user-images.githubusercontent.com/52528841/170788600-28da87b7-ce63-4924-a83c-4135e08f7724.png)
Does this help or do you mean something different? if you like you can take a screenshot and change it in gimp, then i can imagine it a bit better

**Reply by chip33** [reply]
Hi, 
I do know about that option, I just meant an option to hide the Title Bar would be good, and it would also save a little space height wise and let you add an extra item or 2 for the same Row Height.
I also think @TransLucida suggestions are really good too. He or She has a great eye for UI design.

Allow user to hide the search bar
Allow user to hide the bottom icons
Make bottom icons much smaller in comparison to the other icons
Remove or allow user to hide the item count at the bottom
Increase the margins around the menu so the icons don't stay too close to the edges
Allow user to let the program check for updates


Great to see this project has so many keen followers :)

**Comment by TransLucida** [comment]
I did this very quickly on Fireworks just to give an idea. It’s a clean interface with good spaces around the icons and between the icons and the edges of the menu. I like clean interfaces and in my experience designing UI most people seem to have difficult identifying things that are right in front of their faces when there are too many elements on display, or too many options.

Even though I didn’t mention it in my list I added the down-right arrows, which I understand it may not be suitable for most people, also I like the current idea of being able to move the menu around. For me it doesn’t always work as whenever I change my monitor resolution it shows up in a different place – which I imagine it’s because its X, Y are related to the top left of the screen.

The hamburger menu is being used as a placeholder because I just couldn’t find a nice folder icon, but the arrangement would be to let the users decide whether they would want to show everything in alphabetical order (both folders and files) or choose to keep folders grouped at either top or bottom.

The idea of allowing the user to create divisions I can imagine would open a can of worms since the user wouldn’t be able to just use a folder as menu but create an entire menu from scratch which I understand it may not be the scope of the project.

The top of the menu, which could be either something the user typed in the settings, or the name of the root folder doubles as a hot area. When the user clicks/taps this area it becomes a search box, hence the icon next to it.

Some quick action icons could be added to the bottom, but in order to achieve a clean interface users should be allowed to hide them altogether, leaving all options to a right-click standard windows menu.

I hope I didn’t rough any feathers with my suggestions. I really like the project, I use it everyday and I can only expect it to get better.

________________________________

**Comment by Hofknecht** [comment]
@TransLucida 
All ideas processed. Converted to Q/A to mark as answered.

Thank you very much!

If you have further questions, suggestions or ideas, please open further discussions or issues. We would be very happy about that!
